lib/api/src/js/macros.rs

```diff
@@ -0,0 +1,32 @@
+/// Assert that a type does **not** implement a set of traits.
+macro_rules! assert_not_implemented {
+    ($t:ty : !$first:ident $(+ !$rest:ident)*) => {
```

Comment:
Just a small nit: `assert_not_implemented!(Instance: !Send + !Sync);` sounds like a double negative. so one might initially expect `Instance` to *not* implement `!Send`, i.e. that `Instance` should be `Send`. I don't have suggestions for alternate syntax though.